src/store.ts

```diff
@@ -121,8 +138,12 @@ const store: TypedVuexStore = new TypedVuexStore({
     setOnboardingData(state: VuexStoreState, onboardingData: AppOnboardingData) {
       state.onboardingData = onboardingData;
     },
+    setOrderByNewest(state: VuexStoreState, orderByNewest: boolean) {
+      state.orderByNewest = orderByNewest;
+      state.semesters = sorted(state.semesters, orderByNewest);
```

Comment:
We ideally want to keep `setXYZ` as simple as possible.

You should move this into a check below, i.e.
```
if (payload.type === 'setOrderByNewest') {
    store.commit('setSemesters', sorted(state.semesters, orderByNewest));
}
```

(lmk if you don't know what I mean)